### PR TITLE
Make `Context::new()` public (close #490)

### LINKED ISSUE #490: Please make `Context::new()` public
Say we have three actors, each needs the address of the other two:
```rust
struct Actor1 {
    actor2: Addr<Actor2>,
    actor3: Addr<Actor3>,
}

struct Actor2 {
    actor1: Addr<Actor1>,
    actor3: Addr<Actor3>,
}

struct Actor3 {
    actor1: Addr<Actor1>,
    actor2: Addr<Actor2>,
}
```

If we want to actually launch all these actors, we'd have to do use an abomination like that:
```rust
Actor1::create(|ctx1| {
    let mut actor3_addr = None;
    let actor2 = Actor2::create(|ctx2| {
        let actor3 = Actor3 {
            actor1: ctx1.address(),
            actor2: ctx2.address(),
        }.start();
        actor3_addr = Some(actor3.clone());
        Actor2 {
            actor1: ctx1.address(),
            actor3
        }
    });
    let actor3 = actor3_addr.unwrap();
    Actor1 {
        actor2,
        actor3,
    }
});
```

But if `Context::new()` was public, we could do:
```rust
let ctx1 = Context::<Actor1>::new();
let ctx2 = Context::<Actor2>::new();
let ctx3 = Context::<Actor3>::new();

let actor1 = Actor1 {
    actor2: ctx2.address(),
    actor3: ctx3.address(),
};

let actor2 = Actor2 {
    actor1: ctx1.address(),
    actor3: ctx3.address(),
};

let actor3 = Actor3 {
    actor1: ctx1.address(),
    actor2: ctx2.address(),
};

ctx1.run(actor1);
ctx2.run(actor2);
ctx3.run(actor3);
```

Note that this is currently doable with `Context::with_receiver()`:
```rust
let (_, rx) = actix::dev::channel::channel(16);
let ctx1 = Context::<Actor1>::with_receiver(rx);
// ...
```

But this feels hacky and ugly, and it would be much better if we could just use `Context::new()` directly. It's already there - it just needs to be made public [instead of crate public](https://github.com/actix/actix/blob/2cb738a48c8283c25956c660d597734429d58c75/actix/src/context.rs#L86).

### PR DESCRIPTION
**Pull request by idanarye**
## PR Type

Feature

## PR Checklist
Check your PR fulfills the following:

- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [ ] Format code with the latest stable rustfmt

## Overview

Closes #490

### CODE REVIEW

**Comment by fakeshadow** [comment]
Maybe a short doc example to show how to use this directly to start an actor?

File: actix/CHANGES.md [hunk 1]
```diff
@@ -5,6 +5,9 @@
 * Add `fut::try_future::ActorTryFuture`. [#419]
 * Add `fut::try_future::ActorTryFutureExt` trait with `map_ok`, `map_err` and `and_then` combinator. [#419] 
 
+### Changed
+- Make `Context::new` public.
+
 [#419]: https://github.com/actix/actix/pull/419
```
Comment (JohnTitor) [inline actix/CHANGES.md:12]:
```suggestion
- Make `Context::new` public. [#491]

[#419]: https://github.com/actix/actix/pull/419
[#491]: https://github.com/actix/actix/pull/491
```

**Review by JohnTitor — APPROVED** [review]
LGTM with the below nit and @fakeshadow's comment.

File: actix/CHANGES.md [hunk 2]
```diff
@@ -5,7 +5,11 @@
 * Add `fut::try_future::ActorTryFuture`. [#419]
 * Add `fut::try_future::ActorTryFutureExt` trait with `map_ok`, `map_err` and `and_then` combinator. [#419] 
 
+### Changed
+- Make `Context::new` public. #[491]
```
Comment (JohnTitor) [inline actix/CHANGES.md:9]:
```suggestion
- Make `Context::new` public. [#491]
```

**Comment by JohnTitor** [comment]
Thanks!

**Pull request merged**